### PR TITLE
eth/tracers: expose contextual infos (block hash, tx hash, tx index)

### PR DESCRIPTION
**Pull request by karalabe**

### CODE REVIEW

File: eth/tracers/tracer.go [hunk 1]
```diff
@@ -312,10 +312,18 @@ type Tracer struct {
 	reason    error  // Textual reason for the interruption
 }
 
+// Context contains some contextual infos for a transaction execution that is not
+// available from within the EVM object.
+type Context struct {
+	BlockHash common.Hash // Hash of the block the tx is contained within (zero if dangling tx or call)
```
Comment (s1na) [inline eth/tracers/tracer.go:318]: I probably missed the memo, just wondering why you exported the fields?

Comment (karalabe) [reply eth/tracers/tracer.go:318]: It's part of the API (`tracers.New`).

**Review by s1na — APPROVED** [review]
Looks good

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**